### PR TITLE
Add improved logic, with fallbacks.

### PR DESCRIPTION
**Pull request by ryan-blake-affinity**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] non breaking change (non-breaking change which is not issue related)
- [ ] Security related (updates which are security related)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/sagepay/src/Plugin/Payment/Method/SagePayBase.php [hunk 1]
```diff
@@ -114,16 +114,26 @@ public function needCard() {
    * {@inheritdoc}
    */
   public function setPaymentExecutionResult($response, $request) {
+
+    // Set some default values for the payment execution result.
     $url = '';
     $status = 200;
```
Comment (SOMERSOFTAndrewCrook) [inline modules/sagepay/src/Plugin/Payment/Method/SagePayBase.php]: Can we change this to 
```php
$status = Response::HTTP_OK;
```
?

File: modules/sagepay/src/Plugin/Payment/Method/SagePayBase.php [hunk 2]
```diff
@@ -114,16 +114,26 @@ public function needCard() {
...
     $headers = [];
+
+    // Check if instance of \Omnipay\SagePay\Message\ServerAuthorizeResponse.
     /** @var \Drupal\payment\Response\Response $response */
```
Comment (SOMERSOFTAndrewCrook) [inline modules/sagepay/src/Plugin/Payment/Method/SagePayBase.php]: Given the check on the next line, perhaps better to duplicate the line and move this line to a better place?

**Review by SOMERSOFTAndrewCrook — CHANGES_REQUESTED** [review]
Two small changes.

File: modules/sagepay/src/Plugin/Payment/Method/SagePayBase.php (hunk 2 above)
Comment (ryan-blake-affinity) [reply modules/sagepay/src/Plugin/Payment/Method/SagePayBase.php]: Yeah can do

**Review by SOMERSOFTAndrewCrook — APPROVED** [review]

**Pull request merged**